open-banking-accelerator/components/com.wso2.openbanking.accelerator.common/src/main/java/com/wso2/openbanking/accelerator/common/util/JWTUtils.java

```diff
@@ -120,4 +135,101 @@ public static boolean validateJWTSignature(String jwtString, String jwksUri, Str
         return true;
     }
 
+    /**
+     *Validates the signature of a given JWT against a given public key.
+     *
+     * @param signedJWT the signed JWT to be validated
+     * @param publicKey the public key that is used for validation
+     * @param algorithm the algorithm expected to have signed the jwt
+     * @return true if signature is valid else false
+     * @throws NoSuchAlgorithmException if the given algorithm doesn't exist
+     * @throws InvalidKeySpecException if the provided key is invalid
+     * @throws JOSEException if an error occurs during the signature validation process
+     */
+    @Generated(message = "Excluding from code coverage as KeyFactory does not initialize")
+    public static boolean validateJWTSignature(SignedJWT signedJWT, String publicKey, String algorithm) throws
+            NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+
+        byte[] publicKeyData = Base64.getDecoder().decode(publicKey);
+        X509EncodedKeySpec spec = new X509EncodedKeySpec(publicKeyData);
+        KeyFactory kf = KeyFactory.getInstance(algorithm);
+        RSAPublicKey rsapublicKey = (RSAPublicKey) kf.generatePublic(spec);
```

Comment:
Shall we check the possibility of using [this](https://www.javadoc.io/doc/com.nimbusds/nimbus-jose-jwt/5.1/com/nimbusds/jose/jwk/RSAKey.html#parse-java.lang.String-) method?

Reply:
There is a issue when generating RSAKey using string stored in db.